cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -0,0 +1,165 @@
+
+  auto graph = mgp::Graph(memgraph_graph);
+
+  std::map<mgp::Node, mgp::Node> old_new_node_mirror;
+  for (auto node : nodes) {
+    if (CheckIfStandin(node, standin_nodes)) {
+      continue;
+    }
+    mgp::Node new_node = graph.CreateNode();
+
+    for (auto label : node.Labels()) {
+      new_node.AddLabel(label);
+    }
+
+    for (auto prop : node.Properties()) {
+      if (skip_props.Empty() || !skip_props_searchable.contains(mgp::Value(prop.first))) {
+        new_node.SetProperty(prop.first, prop.second);
+      }
+    }
+    old_new_node_mirror.insert({node, new_node});
+    InsertCloneNodesRecord(memgraph_graph, result, memory, node.Id().AsInt(), new_node.Id().AsInt());
+  }
+
+  for (auto rel : rels) {
+    mgp::Relationship new_relationship =
+        graph.CreateRelationship(GetStandinOrCopy(standin_nodes, rel.From(), old_new_node_mirror),
+                                 GetStandinOrCopy(standin_nodes, rel.To(), old_new_node_mirror), rel.Type());
+    for (auto prop : rel.Properties()) {
+      if (skip_props.Empty() || !skip_props_searchable.contains(mgp::Value(prop.first))) {
+        new_relationship.SetProperty(prop.first, prop.second);
+      }
+    }
+  }
+}
+
+void Refactor::CloneSubgraphFromPaths(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result,
+                                      mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  try {
+    const auto paths = arguments[0].ValueList();
+    const auto config_map = arguments[1].ValueMap();
+
+    std::unordered_set<mgp::Node> distinct_nodes;
+    std::unordered_set<mgp::Relationship> distinct_relationships;
+    for (auto path_value : paths) {
+      auto path = path_value.ValuePath();
+      for (size_t index = 0; index < path.Length(); index++) {
+        distinct_nodes.insert(path.GetNodeAt(index));
+        distinct_relationships.insert(path.GetRelationshipAt(index));
+      }
+      distinct_nodes.insert(path.GetNodeAt(path.Length()));
+    }
+    std::vector<mgp::Node> nodes_vector{distinct_nodes.begin(), distinct_nodes.end()};
+    std::vector<mgp::Relationship> rels_vector{distinct_relationships.begin(), distinct_relationships.end()};
+    CloneNodesAndRels(memgraph_graph, result, memory, nodes_vector, rels_vector, config_map);
+
+  } catch (const std::exception &e) {
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+
+void Refactor::CloneSubgraph(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  try {
+    const auto nodes = arguments[0].ValueList();
+    const auto rels = arguments[1].ValueList();
+    const auto config_map = arguments[2].ValueMap();
+
+    std::unordered_set<mgp::Node> distinct_nodes;
+    std::unordered_set<mgp::Relationship> distinct_rels;
+
+    for (auto node : nodes) {
+      if (node.IsNode()) {
+        distinct_nodes.insert(node.ValueNode());
+      }
```

Comment:
These are 100% nodes so no need for if here. In `CallCustomProcedure` function in memgraph you can check there we are checking if types are correct

Reply:
sorry, working with config maps got me used to always checking what gets passed😂